### DISCUSSION TITLE
Carla Newbie: More documentation available?  + feature proposal

### DISCUSSION THREAD

**Discussion by HarryR2** [falkTX/Carla#1657] [Q&A]
Hi everybody !!
I'm kind of a midi (and so Carla) newbie :-)

I recently discovered Carla and I like it very much.

What I'm missing is a documentation which goes a little deeper than
https://kx.studio/Documentation:Manual:Carla

For example : How does "Midi learn" work ?
When I click it and try whatever with my midi-keyboard nothing happens ..

What I'm missing (which leads to a feature proposal) is something like a converting values of midi events.

I'll try to explain what I mean with a simple and maybe not really useful example.
Think of a controller X that sends values between 1 and 127.

The plugin has a control for wet and a control for dry volume (as output).

I could use X to control the wet output of a plugin (1 -> 127).
If I'd  use X too for the dry output, but  reverse (1 becomes 127, 127 becomes 1)
sending midi events from a keyboard-fader would lead to a smooth mix of both outputs.

So, there just has to be one further option to reverse incoming midi value (which is my proposal).

I hope I wrote my proposal in an understandable way (sorry English is not my first language).

If there's already an alternative way, better than my proposal, please tell me !

Best regards
Harry

**Comment by rdica** [comment]
+1 on need for better documentation.

I just came across an undocumented feauture using carla-patchbay as a plugin. The plugin itself seems to use a few midi CCs for itself and doesn't pass them to the plugins added to the patchbay. Nowhere is this documented, nor is there any means to configure this.
CC numbers 7, and 10 seem to be reserved for use by the carla-patchbay plugin itself. Behaviour experienced on all midi channels.

While I cannot think of a use case where this functionality would be useful, I'm sure there is a use case, so I'm not calling it a bug, but I was annoyed at wasting hours of time trying to troubleshoot when the fault wasn't mine in my configuration but that of unknown/undocumented behaviour.
